docs/README.md

```diff
@@ -168,6 +168,65 @@ Rationale:
 
 For more details about leveraging Mermaid diagrams in Docusaurus, see [here](https://docusaurus.io/docs/markdown-features/diagrams).
 
+### Documents vs Pages
+
+*Documents* and *Pages* are distinct concepts in Docusaurus.
+For an overview of each, see: [Documents](https://docusaurus.io/docs/docs-introduction) and [Pages](https://docusaurus.io/docs/creating-pages).
+
+Some primary distinctions:
+
+1. *Documents* live under `docs` and `versioned_docs`. *Pages* live under `src/pages`.
+1. *Documents* are versioned. *Pages* are not.
+
+### The Sidebar
+
+For an overview of how to configure sidebars in Docusaurus, see [here](https://docusaurus.io/docs/sidebar).
+
+The site's "current" version sidebar is configured via `sidebars.ts`.
+
+Sidebars for other versions are configured via `versioned_sidebars/version-<version-id>.json`.
+
+- Versioned sidebars do not yet support JS/TS file formats.
+  See <https://github.com/facebook/docusaurus/issues/10407>.
+
+Note that sidebars are configured for documents under `docs` and `versioned_docs`; they do not apply to unversioned *Pages*.
+
+### Documentation Versioning
+
+For an overview of Docusaurus's versioning functionality, see [here](https://docusaurus.io/docs/versioning).
+
+We currently offer versioned documentation for each of our supported major versions.
+This documentation is intended to be kept up-to-date with the most recent release of each major version series.
+
+For now, this means we publish documentation (including generated API documentation) for versions `1.x` and `2.x`.
+
+- We also support generating API documentation for the local repo code in local development only.
+  See [Local API docs build](#local-api-docs-build), but these are not intended to be published.
+
+#### Why Only Major Versions?
+
+We aim to keep the number of concurrently maintained site versions minimized, for a number of reasons:
+
+1. Developer overhead - the more versions of the docs we offer, the more work we take on to keep documentation for supported versions up to date.
+2. User overhead - the more versions we offer, the more users have to think about when they engage with our website.
+3. Build performance - Docusaurus's build isn't fast. The more versions we add to our suite, the longer our build times become.
+
+#### Updating the Site Version
+
+These steps are based on Docusaurus's tutorial [here](https://docusaurus.io/docs/versioning#tutorials).
+
+Note: in order for API documentation to be available for the new version, you will need to ensure that API model artifacts are being published for the new major version series.
```

Comment:
@zhenmichael I'd like to document the steps for setting up API model artifacts publishing for a new release branch, if possible. Hoping you can help me fill that in 🙂

Reply:
Looks good overall! The pubishing pipeline should automatically generate new artifacts in the blob storage the first time a new major version release version is run. As long as the config file is updated the download-apis step should have access the correct artifacts.